### PR TITLE
QR code animation when keyboard is up, iOS only

### PR DESCRIPTION
**Pull request by amanalili**

### CODE REVIEW

**Comment by ahujsak** [comment]
@amanalili  there are conflicts with this branch, can you fix and resubmit

File: src/modules/UI/components/WalletListModal/WalletListModal.ui.js [hunk 1]
```diff
@@ -20,8 +20,8 @@ export default class WalletListModal extends Component<Props> {
     const top = this.props.topDisplacement ? this.props.topDisplacement : 38
     return (
       <Animatable.View style={[b(), styles.topLevel, {position: 'absolute', top: top, height: (platform.deviceHeight - platform.toolbarHeight - platform.footerHeight)}]}
-        animation='fadeInDown'
-        duration={100} >
+        animation='fadeInUp'
```
Comment (wolverineks) [inline src/modules/UI/components/WalletListModal/WalletListModal.ui.js]: what does this change do?

File: src/modules/UI/scenes/Request/Request.ui.js [hunk 2]
```diff
@@ -195,4 +221,59 @@ export default class Request extends Component {
...
+  _animateQRCodeOnShow (event) {
+    Animated.timing(this.state.animationQrSize, {
+      duration: event.duration,
+      toValue: platform.deviceHeight / 4.3,
+    }).start()
+    Animated.timing(this.state.animationPushUpSize, {
+      duration: event.duration,
+      toValue: 60,
+    }).start()
+    // Animated.spring(this.state.animationQrSize,{
```
Comment (wolverineks) [inline src/modules/UI/scenes/Request/Request.ui.js]: Please remove any dead code

File: src/modules/UI/scenes/Request/Request.ui.js [hunk 3]
```diff
@@ -195,4 +221,59 @@ export default class Request extends Component {
...
+    this._animateQRCodeOnShow(event)
+  }
+  keyboardWillHide (event) {
+    this.setState({
+      keyboardUp: false
+    })
+    this._animateQRCodeOnHide(event)
+  }
+
+  _animateQRCodeOnShow (event) {
```
Comment (wolverineks) [inline src/modules/UI/scenes/Request/Request.ui.js]: Please refrain from prefixing methods with `_`

**Comment by wolverineks** [comment]
@amanalili Please rebase or merge from master more frequently to avoid conflicts

File: src/modules/UI/scenes/Request/Request.ui.js [hunk 4]
```diff
@@ -67,6 +74,16 @@ export default class Request extends Component {
     .catch((e) => console.log(e))
   }
 
+  componentWillMount () {
+    this.keyboardWillShowListener = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow.bind(this))
```
Comment (wolverineks) [inline src/modules/UI/scenes/Request/Request.ui.js]: Android does not have `keyboardWillShow` or `keyboardWillHide`

File: android/app/build.gradle [hunk 5]
```diff
@@ -138,7 +138,7 @@ android {
 
 dependencies {
     compile project(':react-native-picker')
-    compile project(':react-native-smart-splashscreen')
+    compile project(':react-native-smart-splash-screen')
```
[the remark this answers is not in the document]
Comment (amanalili) [reply android/app/build.gradle]: Makes errors on android when that is not renamed

File: android/app/src/main/java/co/edgesecure/wallet/MainApplication.java [hunk 6]
```diff
@@ -3,8 +3,8 @@
 import android.app.Application;
 
 import com.facebook.react.ReactApplication;
-import com.beefe.picker.PickerViewPackage;
```
[the remark this answers is not in the document]
Comment (amanalili) [reply android/app/src/main/java/co/edgesecure/wallet/MainApplication.java]: No idea what that is, never used that.

File: android/settings.gradle [hunk 7]
```diff
@@ -1,8 +1,8 @@
 rootProject.name = 'co.edgesecure.wallet'
+include ':react-native-smart-splash-screen'
```
[the remark this answers is not in the document]
Comment (amanalili) [reply android/settings.gradle]: Makes errors on android when that is not renamed

File: ios/edge.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -5,7 +5,6 @@
 	};
 	objectVersion = 46;
 	objects = {
-
 /* Begin PBXBuildFile section */
```
[the remark this answers is not in the document]
Comment (amanalili) [reply ios/edge.xcodeproj/project.pbxproj]: Errors upon merge from the last

File: src/modules/UI/components/WalletListModal/WalletListModal.ui.js (hunk 1 above)
Comment (amanalili) [reply src/modules/UI/components/WalletListModal/WalletListModal.ui.js]: Another animation change, sorry did not feature branched it. Did that before I was briefed

File: src/modules/UI/scenes/Request/Request.ui.js (hunk 3 above)
Comment (amanalili) [reply src/modules/UI/scenes/Request/Request.ui.js]: noted

File: src/modules/UI/scenes/Request/Request.ui.js (hunk 2 above)
Comment (amanalili) [reply src/modules/UI/scenes/Request/Request.ui.js]: noted

File: src/modules/UI/scenes/Request/Request.ui.js (hunk 4 above)
Comment (amanalili) [reply src/modules/UI/scenes/Request/Request.ui.js]: I know, no android animation first

**Comment by amanalili** [comment]
Pushed comment fixes and flow fixes.

**Comment by amanalili** [comment]
Pushed commits to make the native code exactly the same as master. For the comment to be resolve.

**Comment by paullinator** [comment]
Art. Please resolve the conflict.

**Comment by paullinator** [comment]
Art. Your branch looks to be very stale compared to master. Please merge master into art-animation-qrcode, resolve any conflicts. Fully test your code on iOS and Android, and resubmit the PR.

File: src/modules/UI/components/QRCode/QRCode.ui.js [hunk 9]
```diff
@@ -1,21 +1,27 @@
...
-          style={styles.qrCode}
-          value={this.props.value}
-          bgColor={styles.qrCodeBackground.color}
-          fgColor={styles.qrCodeForeground.color}
-          size={platform.deviceHeight / 4} />
-      </View>
+      <Animated.View style={[ styles.qrCodeBorder, {marginBottom: this.props.animationPushUpSize}]}>
+        <Animated.Image
+          style={{width: this.props.animationQrSize, height: this.props.animationQrSize}}
+          source={{uri: qrcode(this.props.value)}}
```
Comment (paullinator) [inline src/modules/UI/components/QRCode/QRCode.ui.js]: I think this is causing your slow performance on animation. The qrcode() function might be pretty slow. You should only reprocess the QRcode if the user types in the textfield or if the user entered value changes. During the animation, you should only take the resulting data from qrcode() and animate it. The way you have this coded, you'll call the function qrcode() on every frame which I'm sure will be expensive. Please refactor to fix this performance issue

**Pull request closed without merging**